node/node.go

```diff
@@ -589,6 +593,60 @@ func (n *Node) SignBlobs(blobs []*core.BlobMessage, referenceBlockNumber uint) (
 	return signatures, nil
 }
 
+// ValidateBlobHeadersRoot validates the blob headers root hash
+// by comparing it with the merkle tree root hash of the blob headers.
+// It also checks if all blob headers have the same reference block number
+func (n *Node) ValidateBlobHeadersRoot(ctx context.Context, blobHeaderHashes [][32]byte, root [32]byte) error {
+	leafs := make([][]byte, 0)
+	referenceBlockNumber := uint32(0)
```

Comment:
We should be comparing the `referenceBlockNumber`s againts the one in the batch being attested. What if we just passed in the entire `BatchHeader` and used its reference block number here instead of just passing in the merkle root?

Reply:
That makes sense!